### PR TITLE
Use native PulseAudio protocol for volume module

### LINKED ISSUE #302: pipewire audio volume incompatible
The volume module seems to be incompatible with pipewire. With the pulseaudio.DefaultSink() I get just an error, although pipewire-pulse is installed. 

And with the alsa.DefaultSink() I get completely wrong values. Right now the Volume is 1%, but according to pamixer --get-volume it's set to 23%....

Any idea how to fix this? Is this a change in pipewire's pulse implementation?

### PR DESCRIPTION
**Pull request by WGH-**
Most importantly, PipeWire does not have module-dbus-protocol implemented at all[1], so barista didn't work with PipeWire.

Note that PipeWire does implement native PulseAudio protocol - and that's how most applications interface with PipeWire, using it for streaming audio, sink/source control (pavucontrol), etc. PipeWire's own native protocol is not frequently used as of yet.

Secondly, module-dbus-protocol is frequently not loaded by default on PulseAudio, necessitating explicit "pactl load-module module-dbus-protocol".

[1] https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/1127

Fixes #302

### CODE REVIEW

**Comment by WGH-** [comment]
I'm using this code on my machine with PipeWire, and I did some basic tests with PulseAudio in Docker.

The only problem I have encountered so far is that it seemingly failed to connect on logon, and I had to `i3-msg restart`. I'll investigate why (PulseAudio/PipeWire is supposed to be socket-activated), and either try to fix it here or in a follow-up PR.

File: modules/volume/pulseaudio/pulse.go [hunk 1]
```diff
@@ -115,131 +70,142 @@ func Source(sourceName string) volume.Provider {
...
+	err = client.Request(&proto.GetSourceInfo{SourceIndex: proto.Undefined, SourceName: deviceName}, &repl)
 	if err != nil {
-		return volume.Volume{}, err
+		return
 	}
+	return makeVolume(repl.ChannelVolumes, repl.BaseVolume, repl.Mute, &sourceController{client, deviceName}), nil
+}
 
+func makeVolume(channelVolumes proto.ChannelVolumes, baseVolume proto.Volume, mute bool, controller volume.Controller) volume.Volume {
+	maxVol := int64(baseVolume)
```
Comment (WGH-) [inline modules/volume/pulseaudio/pulse.go:148]: Okay, I've just noticed that this part is wrong. 100% volume should be `PA_VOLUME_NORM` (0x10000), not base volume.

Comment (WGH-) [reply modules/volume/pulseaudio/pulse.go]: Fixed

**Comment by soumya92** [comment]
:cowboy_hat_face: Let's see how this goes

**Comment by WGH-** [comment]
> The only problem I have encountered so far is that it seemingly failed to connect on logon, and I had to i3-msg restart. I'll investigate why (PulseAudio/PipeWire is supposed to be socket-activated), and either try to fix it here or in a follow-up PR.

FYI, it seems it was fixed by #322.

Apparently, during the login, barista starting and connecting to PulseAudio/PipeWire races with devices appearing.

**Pull request merged**